### PR TITLE
Add structured logger to epidata, and log metadata generation

### PR DESCRIPTION
**Pull request by benjaminysmith**
- Add a structured logger to the epidata repo
- Add logging for time taken to generate and update metadata

Progress on #403

Note that this needs structlog installed before running since this adds a new dependency.

The logger is a fork of the logger in delphi_utils. We can unify these if we extract delphi utils into a standalone package.

### CODE REVIEW

File: src/acquisition/covidcast/covidcast_meta_cache_updater.py [hunk 1]
```diff
@@ -3,30 +3,38 @@
...
 
 
 def main(args, epidata_impl=Epidata, database_impl=Database):
   """Update the covidcast metadata cache.
 
   `args`: parsed command-line arguments
   """
+  logger = get_structured_logger(
+      "metadata_cache_updater",
+      filename=args.log_file)
```
Comment (melange396) [inline src/acquisition/covidcast/covidcast_meta_cache_updater.py:28]: i take it this means the log file argument is required...  is there maybe a way to let it default to stdout or somehing?

File: src/acquisition/covidcast/covidcast_meta_cache_updater.py [hunk 2]
```diff
@@ -44,13 +52,22 @@ def main(args, epidata_impl=Epidata, database_impl=Database):
 
   # update the cache
   try:
+    metadata_update_start_time = time.time()
     database.update_covidcast_meta_cache(metadata)
+    metadata_update_interval_in_seconds = time.time() - metadata_update_start_time
```
Comment (melange396) [inline src/acquisition/covidcast/covidcast_meta_cache_updater.py:57]: this should always be very quick, but it doesnt hurt to measure it anyway

**Review by melange396 — APPROVED** [review]
LGTM

**Comment by krivard** [comment]
Waiting to merge until we have the bandwidth to install structlog on prod

File: src/acquisition/covidcast/covidcast_meta_cache_updater.py (hunk 2 above)
Comment (benjaminysmith) [reply src/acquisition/covidcast/covidcast_meta_cache_updater.py:57]: acknowledged

File: src/acquisition/covidcast/covidcast_meta_cache_updater.py (hunk 1 above)
Comment (benjaminysmith) [reply src/acquisition/covidcast/covidcast_meta_cache_updater.py:28]: If the arg is empty/none then the logger will default to stdout.

**Comment by benjaminysmith** [comment]
Structlog is in but lets wait to merge until we have the regular system functional again (don't want to introduce additional risk)

**Comment by krivard** [comment]
Primary is relatively healthy; merged

**Pull request merged**